doc/source/dataset.rst

```diff
@@ -0,0 +1,50 @@
+Ray Datasets
```

Comment:
can we put this in a folder e.g., `doc/source/dataset/index.rst`

This is what we did with serve for example.

Reply:
+1 here